manage_arkime/core/capacity_planning.py

```diff
@@ -90,4 +96,165 @@ def get_ecs_sys_resource_plan(instance_type: str) -> EcsSysResourcePlan:
+
+@dataclass
+class DataNodesPlan:
+    count: int
+    type: str
+    vol_size: int # in GiB
+
+    def __equal__(self, other):
+        return (self.count == other.count and self.type == other.type
+                and self.vol_size == other.vol_size)
+
+    def to_dict(self) -> Dict[str, str]:
+        return {
+            "count": self.count,
+            "type": self.type,
+            "volSize": self.vol_size
+        }
+    
+@dataclass
+class MasterNodesPlan:
+    count: int
+    type: str
+
+    def __equal__(self, other):
+        return (self.count == other.count and self.type == other.type)
+
+    def to_dict(self) -> Dict[str, str]:
+        return {
+            "count": self.count,
+            "type": self.type
+        }
+
+@dataclass
+class OSDomainPlan:
+    data_nodes: DataNodesPlan
+    master_nodes: MasterNodesPlan
+
+    def __equal__(self, other):
+        return (self.data_nodes == other.dataNodes
+                and self.master_nodes == other.masterNodes)
+
+    def to_dict(self) -> Dict[str, str]:
+        return {
+            "dataNodes": self.data_nodes.to_dict(),
+            "masterNodes": self.master_nodes.to_dict()
+        }
+    
+def _get_storage_per_replica(expected_traffic: float, spi_days: int) -> float:
+    """
+    Predict the required OpenSearch domain storage for each replica, in GiB
+
+    expected_traffic: traffic volume to the capture nodes, in Gbps
+    spi_days: the number of days to retain the SPI data stored in the OpenSearch Domain
+    """
+    return (spi_days * 24 * 60 * 60) * expected_traffic/8 * MAGIC_FACTOR
+
+def _get_total_storage(expected_traffic: float, spi_days: int, replicas: int) -> float:
+    """
+    Predict the total required OpenSearch domain storage, in GiB
+
+    expected_traffic: traffic volume to the capture nodes, in Gbps
+    spi_days: the number of days to retain the SPI data stored in the OpenSearch Domain
+    replicas: the number of replicas to have of the data
+    """
+    return _get_storage_per_replica(expected_traffic, spi_days) * replicas
+
+def _get_data_node_plan(total_storage: float) -> DataNodesPlan:
+    """
+    Per the OpenSearch Service limits doc [1], you can have a maximum of 10 T2/T3 data nodes or 80 of other types by
+    default.  You can raise this limit up to 200.  To keep things simple, we will assume if the user needs more storage
+    than 80 of the largest instance type can provide, they'll bump the limit out of band and just keep getting more of
+    that largest instance type.
+
+    There's also an apparent incentive to have more, smaller nodes than fewer, larger nodes [2]
+
+    [1] https://docs.aws.amazon.com/opensearch-service/latest/developerguide/limits.html
+    [2] https://github.com/arkime/aws-aio/issues/56#issuecomment-1563652060
+
+    total_storage: full stage requirement for all data, including replicas, in GiB
+    """
+
+    if total_storage <= 10 * T3_SMALL_SEARCH.vol_size:
```

Comment:
Great first version, later would like this to be a table that we just scan instead of hard coding instances in the logic.